analytics-kotlin/src/main/java/com/segment/analytics/platform/plugins/StartupQueue.kt

```diff
@@ -0,0 +1,63 @@
+package com.segment.analytics.platform.plugins
+
+import com.segment.analytics.Analytics
+import com.segment.analytics.BaseEvent
+import com.segment.analytics.platform.Plugin
+import com.segment.analytics.System
+import sovran.kotlin.Subscriber
+import java.util.Queue
+import java.util.concurrent.ConcurrentLinkedQueue
+import java.util.concurrent.atomic.AtomicBoolean
+
+class StartupQueue(): Plugin, Subscriber {
+    override val type: Plugin.Type = Plugin.Type.Before
+    override val name: String = "Segment_StartupQueue"
+    override lateinit var analytics: Analytics
+
+    private val maxSize = 1000
+    private val started: AtomicBoolean = AtomicBoolean(false)
+    private val queuedEvents: Queue<BaseEvent> = ConcurrentLinkedQueue()
+
+    override fun setup(analytics: Analytics) {
+        super.setup(analytics)
+        analytics.store.subscribe(
+            subscriber = this,
+            stateClazz = System::class,
+            initialState = true,
+            handler = this::runningUpdate
+        )
+    }
+
+    override fun execute(event: BaseEvent): BaseEvent? {
+        if (!started.get()) {
+            analytics.log("PRAY-message-queuing")
```

Comment:
remove the log's, or take your name out of them. :D

Reply:
whooops 😅